controllers/database_controller.go

```diff
@@ -68,11 +68,23 @@ func (r *DatabaseReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 
 	log.Info("Database CR Status: " + util.ToString(database.Status))
 
-	spec := database.Spec
-	server := spec.NDB
-	ndbClient := ndbclient.NewNDBClient(server.Credentials.LoginUser, server.Credentials.Password, server.Server)
-
-	// log.Info(fmt.Sprintf("Finalizers: %v", database.Finalizers))
+	NDBInfo := database.Spec.NDB
+	secretName := NDBInfo.CredentialSecret
+	username, err := util.GetDataFromSecret(ctx, r.Client, secretName, req.Namespace, ndbv1alpha1.SECRET_DATA_KEY_USERNAME)
```

Comment:
Rather than doing this, do you think we can add one function that would extract all the necessary information in one stretch? 
For instance, we are extracting information one by one: first username, then password, then ca-cert. Seems like this can be moved to a single function.

Reply:
Extracted the function to return a map of the secret data and created a helper method to fetch these credentials from the map in one go. This was a great idea, it reduced the K8s API call to 1 from 3 in every reconcile function call. 😄